hs_core/models.py

```diff
@@ -2076,15 +2076,12 @@ def get_metadata_xml(self, pretty_print=True, include_format_elements=True):
 
     def _get_metadata(self, metatdata_obj):
         """Get resource metadata from content_object."""
-        md_type = ContentType.objects.get_for_model(metatdata_obj)
-        res_type = ContentType.objects.get_for_model(self)
-        self.content_object = res_type.model_class().objects.get(id=self.id).content_object
+
         if self.content_object:
             return self.content_object
         else:
             metatdata_obj.save()
-            self.content_type = md_type
-            self.object_id = metatdata_obj.id
+            self.content_object = metatdata_obj
```

Comment:
Is this lazy loading actually necessary?  content_object is a generic foreign key.  If we could set this field eagerly we could do a prefetch_related on content_object and speed things up some more.

Reply:
I did a quick test and got a speed up of about .5 second  on jeff's my-resource page by adding
```resource_collection = resource_collection.prefetch_related('content_type')```
to get_my_resources_list and then used content_type in place of metadata in the my-resources.html

I have no idea why we're doing this lazy loading and it doesn't seem necessary.  If you agree, we can go ahead and prefetch this data to avoid extra db calls.

Reply:
Even with this lazy loading of 'metadata' we could have prefetched `content_object` and got the same level of performance improvement if any. However, I made the change to set the `content_object` as part of resource creation. I am also now prefetching all metadata elements (title, creators, subjects and dates). This probably helping in page loading time.